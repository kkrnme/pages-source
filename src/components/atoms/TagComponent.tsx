import React, { ReactChild } from "react"
import { Stylable } from "../../utils/Components"
import { Link } from "gatsby"
import { PlainComponent } from "../../utils/PlainComponent"
import styled from "@emotion/styled"
import { ds, ss, cs } from "../../styles"

type TagComponentProps = Stylable & {
  children: ReactChild
}

type TagLinkProps = TagComponentProps & { children: string }

const Plain: PlainComponent<TagComponentProps> = props => (
  <label className={props.className}>{props.children}</label>
)

export const TagComponent = styled(Plain)`
  margin-left: 0.5rem;
  ${ds.padx(".5rem")}
  ${ss.rounded}
  letter-spacing: 0;
  color: ${cs.background};
  border: ${ss.border};
`

export const TagLink: React.FC<TagLinkProps> = props => (
  <TagComponent className="bg-yellow-500">
    <Link to={`/tags/${props.children}/`}>{props.children}</Link>
  </TagComponent>
)
