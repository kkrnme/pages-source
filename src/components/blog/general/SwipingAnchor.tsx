import React from "react"
import { css, keyframes } from "@emotion/core"
import tw from "tailwind.macro"

export const SwipingAnchor: React.FC<{}> = props => (
  <span className="inline-block relative z-0">
    <a
      className="block"
      css={css(tw``, {
        transition: "color .5s ease-in-out",

        color: "inherit",
        textDecoration: "underline #4f6bed 2px",
        "&::before": {
          display: "block",
          content: `""`,
          height: "100%",
          backgroundColor: "#4f6bed00",
          position: "absolute",
          top: 0,
          transition: "background-color 400ms ease-in-out",
          left: 0,
          right: 0,
          zIndex: -1,
          borderRadius: "1px",
        },
        "&:hover": {
          color: "black",
          "&::before": {
            animation: `${anchorWipe} .5s ease-out both`,
            backgroundColor: "#758bf0ff",
          },
        },
      })}
      {...props}
    />
  </span>
)
const anchorWipe = keyframes`
  0% {
    left: 0;
    right: 100%;
  }
  100% {
    left: 0;
    right: 0;
  }
`
export default SwipingAnchor
