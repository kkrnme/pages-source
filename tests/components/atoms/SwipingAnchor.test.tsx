import React from "react"
import renderer from "react-test-renderer"

import { SwipingAnchor } from "../../../src/components/atoms/SwipingAnchor"

describe("SwipingAnchor", () => {
  it("renders correctly: SwipingAnchor", () => {
    const tree = renderer
      .create(<SwipingAnchor to="osaka">osaka</SwipingAnchor>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
