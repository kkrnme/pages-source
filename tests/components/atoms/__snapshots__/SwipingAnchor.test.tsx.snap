// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipingAnchor renders correctly: SwipingAnchor 1`] = `
<span
  className="inline-block relative z-0"
>
  <a
    className="underline-anchor"
    css={
      Object {
        "map": undefined,
        "name": "cqatmp",
        "next": undefined,
        "styles": "
        color: inherit;
        transition: all 0.4s ease-in-out;
        &::before {
          display: block;
          content: \\"\\";
          height: 100%;
          background-color:#63b3ed;;
          position: absolute;
          transition: all 0.4s ease-in-out;
          top: 0;
          left: 0;
          right: 100%;
          z-index: -1;
          border-radius: 1px;
        }
        &:hover {
          color: black;
          &::before {
            left: 0;
            right: 0;
          }
        }
      ",
        "toString": [Function],
      }
    }
    href="osaka"
  >
    osaka
  </a>
</span>
`;
